pandas-stubs/_typing.pyi

```diff
@@ -186,7 +187,10 @@ CompressionDict = dict[str, Any]
 CompressionOptions = Optional[
     Union[Literal["infer", "gzip", "bz2", "zip", "xz", "zstd"], CompressionDict]
 ]
-
+FormattersType = Union[
+    list[Callable], tuple[Callable, ...], Mapping[Union[str, int], Callable]
+]
+FloatFormatType = str | Callable | matplotlib.ticker.EngFormatter
```

Comment:
pandas uses `pandas.io.formats.format.EngFormatter`, so import that, and you remove the `matplotlib` issue.

Reply:
Fixed.